crates/bevy_pbr/src/render/mesh/mod.rs

```diff
@@ -656,6 +616,28 @@ impl MeshPipelineKey {
     }
 }
 
+fn has_skin(layout: &Hashed<InnerMeshVertexBufferLayout>) -> bool {
+    layout.contains(Mesh::ATTRIBUTE_JOINT_INDEX) && layout.contains(Mesh::ATTRIBUTE_JOINT_WEIGHT)
+}
+
+pub fn set_special_mesh_attributes(
```

Comment:
I wouldn't call this method `set_*()` as it doesn't actually modify things in place, and instead returns the modified data. Idk what to call it, but this could use a better name, and maybe a brief doc comment.

Reply:
I opted to remove `set_special_mesh_attributes` and instead refactor the `specialize` impls of ` PrepassPipeline<M>` and `MeshPipeline` to make it easier to spot what needs to change when updating them.